### PR TITLE
Decomp RestorePreDefeatState + 2 more, add enum for RushFlags

### PR DESCRIPTION
**Pull request by JaThePlayer**
Matches 3 functions, nearly matches 2 more, though both require data migration and one has some float issues.
Creates an enum for RushFlags

### CODE REVIEW

File: include/enums.h [hunk 1]
```diff
@@ -3325,4 +3326,10 @@ enum WindowGroupId {
     WINDOW_GROUP_FILE_MENU = 3,
 };
 
+enum RushFlags {
+    RUSHES_NONE       = 0,
+    RUSHES_MEGA_RUSH  = 1,
```
Comment (nanaian) [inline include/enums.h:3331]: Seems to be a disconnect between "RUSHES" and "Rush" here. I'd prefer we call em RUSH_ and the field `rushFlags`

**Comment by BowserSlug** [comment]
🚀🚀🚀 This PR matches 3 functions (+0.02%) on `us`.

**Review by nanaian — CHANGES_REQUESTED** [review]
LGTM otherwise 🎉

File: include/enums.h (hunk 1 above)
Comment (JaThePlayer) [reply include/enums.h]: Renamed to `RUSH_FLAG_MEGA`, `RUSH_FLAG_POWER`, `RUSH_FLAG_NONE` after a discord discussion

**Review by nanaian — APPROVED** [review]

**Comment by ethteck** [comment]
poyfect coyfish

**Pull request merged**